File: jlc_clojure/test/example/belly.rb
# frozen_string_literal: true

require 'jukebox'

# Example tests using jukebox
module ExampleTests
  extend Jukebox

  step 'today is Sunday' do
    |board|
    board
  end

  step 'I ask whether it\'s Friday yet' do
    |board|
    # Write code here that turns the phrase above into concrete actions
    board
  end

  step 'I should be told {string}', {resources: ["kafka/topic-h"]} do
    |board, _string|
    # Write code here that turns the phrase above into concrete actions
    board
  end

  step 'I wait {int} hour' do
    |board, _hours|
    # Write code here that turns the phrase above into concrete actions
    board
  end

  step 'my belly should growl' do
    |board|
    # Write code here that turns the phrase above into concrete actions
    board
  end

  step :before, :after_step, tags: "@tag1 or @tag2 and @rb" do
    |board|
    board
  end

  step :before, :after do
    |board|
    board
  end

  step :before do
    |board|
    board
  end

  require 'jukebox/cukes'
  extend Jukebox::Cukes
  Before do |board|
    board
  end

  step 'a ruby step that fails' do |board|
    pending! # Write code here that turns the phrase above into concrete actions
    board # return the updated board
  end
end

File: jlc_clojure/test/example/belly.clj
(ns example.belly
  (:require [fundingcircle.jukebox :refer [step]]))

(defn- helper-fn-a
  "A test helper fn."
  {:scene/resources [:kafka/topic-a]}
  [])

(defn- helper-fn-b
  "A test helper fn."
  {:scene/resources [:kafka/topic-b]}
  [])

(defn- helper-fn-c
  "A test helper fn."
  {:scene/resources [:kafka/topic-c]}
  [])

(defn- helper-fn-d
  "A test helper fn."
  {:scene/resources [:kafka/topic-d]}
  []
  (helper-fn-c))

(defn- helper-fn-e
  "A test helper fn."
  []
  '(helper-fn-e))

(defn ^:scene/before before-step
  "Called before a scenario."
  {:scene/resources [:kafka/topic-d :kafka/topic-f]
   :scene/tags "@bat"}
  [board _scenario]
  board)

(defn i-have-cukes-in-my-belly
    {:scene/step "I have {int} cukes in my belly"
     :scene/resources [:kafka/topic-e]}
    [board number-of-cukes]
    (helper-fn-a)
    (helper-fn-b)
    (helper-fn-d)
    (helper-fn-e)
    board)

(step "I have this table"
  [board data-table]
  (assoc board :table data-table))

(defn ^:scene/after to-run-after-scenario
  {:scene/tags "@qux"}
  [board _scenario]
  board)

(step :before {:tags "@foo or @bar and @clj"}
  [board scenario]
  board)

(defn the-datafied-table-should-be-foo-col-bar
  {:scene/step "the datafied table should be"}
  [board datafied-table]
  (assert (= (:table board)
             (read-string datafied-table)))
  board)
